### PR TITLE
fix(dpp): [v23 port] non-deterministic fees due to data contract cache

### PR DESCRIPTION
**Pull request by qrayven**
## Issue being fixed or feature implemented

DPP part port of https://github.com/dashevo/platform/pull/444

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: dpp/src/state_repository.rs [hunk 1]
```diff
@@ -59,7 +59,7 @@ pub trait StateRepositoryLike: Send + Sync {
     /// Remove Document
     async fn remove_document(
         &self,
-        contract_id: &Identifier,
+        contract_id: &DataContract,
```
Comment (antouhou) [inline dpp/src/state_repository.rs]: Why the name is `contract_id`, but the type is `DataContract`? Maybe someone who worked on the original implementation can clarify this? @shumkov @jawid-h @shuplenkov

Comment (antouhou) [reply dpp/src/state_repository.rs]: Alright, I had a chat with @shumkov, and it looks like it needs to be renamed to `data_contract`

Comment (qrayven) [reply dpp/src/state_repository.rs]: thanks, I was changing things too fast

**Review by antouhou — DISMISSED** [review]
👍

**Review by antouhou — APPROVED** [review]

**Pull request merged**